src/Microsoft.Identity.Web.AgentIdentities/AgentIdentitiesExtension.cs

```diff
@@ -0,0 +1,117 @@
+            }
+
+            // Register the OidcFic services for agent applications to work.
+            services.AddOidcFic();
+
+            return services;
+        }
+
+
+        /// <summary>
+        /// Updates the options to acquire a token for the agent identity.
+        /// </summary>
+        /// <param name="options">Authorization header provider options.</param>
+        /// <param name="agentApplicationId">The Agent Identity GUID.</param>
+        /// <returns>The updated authorization header provider options.</returns>
+        public static AuthorizationHeaderProviderOptions WithAgentIdentity(this AuthorizationHeaderProviderOptions options, string agentApplicationId)
+        {
+            // It's possible to start with no options, so we initialize it if it's null.
+            if (options == null)
+                options = new AuthorizationHeaderProviderOptions();
+
+            // AcquireTokenOptions holds the information needed to acquire a token for the Agent Identity
+            if (options.AcquireTokenOptions == null)
+                options.AcquireTokenOptions = new AcquireTokenOptions();
+
+            options.AcquireTokenOptions.ForAgentIdentity(agentApplicationId);
+
+            return options;
+        }
+
+        // TODO:make public?
+        private static AcquireTokenOptions ForAgentIdentity(this AcquireTokenOptions options, string agentApplicationId)
+        {
+            if (options.ExtraParameters == null)
+                options.ExtraParameters = new Dictionary<string, object>();
+
+            // Until it makes it way through Abstractions
+            options.ExtraParameters["fmiPathForClientAssertion"] = agentApplicationId;
+
+            // TODO: do we want to expose a mechanism to override the MicrosoftIdentityOptions instead of leveraging
+            // the default configuration section / named options?.
+            options.ExtraParameters["MicrosoftIdentityOptions"] = new MicrosoftEntraApplicationOptions
+            {
+                ClientId = agentApplicationId, // Agent identity Client ID.
+                ClientCredentials = [ new CredentialDescription() {
+                    SourceType = CredentialSource.CustomSignedAssertion,
+                    CustomSignedAssertionProviderName = "OidcIdpSignedAssertion",
+                    CustomSignedAssertionProviderData = new Dictionary<string, object> {
+                        { "ConfigurationSection", "AzureAd" },        // Use the default configuration section name
+                        { "RequiresSignedAssertionFmiPath", true }, // The OidcIdpSignedAssertionProvider will require the fmiPath to be provided in the assertionRequestOptions.
+                    }
+                }]
+            };
+            return options;
+        }
+
+        /*
+         * 
+
+            // Configuration for the Agent application (usual IdWeb/MISE configuration. Can be done programmatically or in appsettings.json).
```

Comment:
do you want to keep the commented code? move to a doc?

Reply:
yes keep for the moment. Will add to doc

Reply:
Removed